internal/controller/cluster/access_control.go

```diff
@@ -401,428 +260,57 @@ type AerospikeUserPasswordProvider interface {
-		if err != nil {
-			return fmt.Errorf(
-				"could not update role %s: %v", roleCreate.name, err,
-			)
-		}
-
-		if err := client.RevokePrivileges(
-			adminPolicy, roleCreate.name, aerospikePrivileges,
-		); err != nil {
-			return fmt.Errorf(
-				"error revoking privileges for role %s: %v", roleCreate.name,
-				err,
-			)
-		}
-
-		logger.Info(
-			"Revoked privileges for role", "role name", roleCreate.name,
-			"privileges", privilegesToRevoke,
-		)
-	}
-
-	if len(privilegesToGrant) > 0 {
-		aerospikePrivileges, err := privilegeStringToAerospikePrivilege(privilegesToGrant)
-		if err != nil {
-			return fmt.Errorf(
-				"could not update role %s: %v", roleCreate.name, err,
-			)
-		}
-
-		if err := client.GrantPrivileges(
-			adminPolicy, roleCreate.name, aerospikePrivileges,
-		); err != nil {
-			return fmt.Errorf(
-				"error granting privileges for role %s: %v", roleCreate.name,
-				err,
-			)
-		}
-
-		logger.Info(
-			"Granted privileges to role", "role name", roleCreate.name,
-			"privileges", privilegesToGrant,
-		)
-	}
-
-	if !reflect.DeepEqual(role.Whitelist, roleCreate.whitelist) {
-		// Set whitelist.
-		if err := client.SetWhitelist(
-			adminPolicy, roleCreate.name, roleCreate.whitelist,
-		); err != nil {
-			return fmt.Errorf(
-				"error setting whitelist for role %s: %v", roleCreate.name, err,
-			)
-		}
-	}
-
-	if role.ReadQuota != roleCreate.readQuota || role.WriteQuota != roleCreate.writeQuota {
-		if err := client.SetQuotas(
-			adminPolicy, roleCreate.name, roleCreate.readQuota, roleCreate.writeQuota,
-		); err != nil {
-			return fmt.Errorf(
-				"error setting quotas for role %s: %v", roleCreate.name, err,
-			)
+			eventType = corev1.EventTypeNormal
+			reason = "UserCreatedOrUpdated"
+			message = fmt.Sprintf("Created or Updated User %s", v.Name)
 		}
-	}
-
-	logger.Info("Updated role", "role name", roleCreate.name)
-	recorder.Eventf(
-		aeroCluster, corev1.EventTypeNormal, "RoleUpdated",
-		"Updated Role %s", roleCreate.name,
-	)
-
-	return nil
-}
-
-// aerospikeUserCreateUpdate creates or updates an Aerospike user.
-type aerospikeUserCreateUpdate struct {
-	// The user's name.
-	name string
 
-	// The password to set. Required for create. Optional for update.
-	password *string
-
-	// The roles to set for the user. These roles and only these roles will be granted to the user after this operation.
-	roles []string
-}
-
-// Execute creates a new Aerospike user or updates an existing one.
-func (userCreate aerospikeUserCreateUpdate) execute(
-	client *as.Client, adminPolicy *as.AdminPolicy, logger logger,
-	recorder record.EventRecorder, aeroCluster *asdbv1.AerospikeCluster,
-) error {
-	user, err := client.QueryUser(adminPolicy, userCreate.name)
-	isCreate := false
-
-	if err != nil {
-		if strings.Contains(err.Error(), userNotFoundErr) {
-			isCreate = true
+	case acl.AerospikeUserDrop:
+		if failed {
+			eventType = corev1.EventTypeWarning
```

Comment:
We can move this eventType out of switch case as this is common across types.
Just check once and initialise as per that